test/e2e/olm_upgrade_test.go

```diff
@@ -0,0 +1,322 @@
+/*
+Copyright 2023.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e
+
+import (
+	"os"
+	"testing"
+
+	. "github.com/onsi/gomega"
+	mcadv1beta1 "github.com/project-codeflare/multi-cluster-app-dispatcher/pkg/apis/controller/v1beta1"
+	rayv1alpha1 "github.com/ray-project/kuberay/ray-operator/apis/ray/v1alpha1"
+
+	batchv1 "k8s.io/api/batch/v1"
+	corev1 "k8s.io/api/core/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	. "github.com/project-codeflare/codeflare-operator/test/support"
+)
+
+var (
+	namespace          *corev1.Namespace
+	namespaceName      string = "test-ns-olmupgrade"
+	serviceAccountName string = "sdk-user"
+)
+
+// Creates a Ray cluster
+func TestMNISTRayClusterUp(t *testing.T) {
+
+	test := With(t)
+	test.T().Parallel()
+	if os.Getenv("RUN_OLM_TESTS") != "true" {
```

Comment:
It may be better to use build tags - https://stackoverflow.com/questions/54165975/go-test-only-run-tests-that-contain-a-build-tag
That way you can specify the OLMupgrade tests when invoking the tests, i.e. `go test -tags olm_upgrade_test ...`

Reply:
The disadvantage of this approach is that the file won't compile if the tag is not enabled.....
Thinking whether it may be better just to remove the condition and specify what tests to run in makefile - to have a separate command there to run upgrade tests.

Reply:
Yes also if we are adding tags, need to adjust to all the e2e tests with build tag to skip these tests running as part of e2e

Reply:
@sutaakar Probably we can use Test grouping for e2e test run like `go test -timeout 30m -v ./test/e2e -run "^TestMNIST.*$"` as all our e2e tests starts with TestMNIST and rename OLM upgrade test to **TestOLMUpgradeRayClusterUp** and **TestOLMUpgradeMnistJobSubmit** . So that we can call these tests specifically in our workflows by removing the condition

Reply:
Or maybe this test can be moved to dedicated folder, i.e. `test/upgrade`.

Reply:
Sure will go with this approach

Reply:
Moved the tests to folder `test/upgrade`.. kept test dependent files in the `test/e2e` as [ReadFile](https://github.com/project-codeflare/codeflare-operator/blob/3bf1ba8cfa31872a28fac26c66914e1551fef94f/test/e2e/support.go#L30) method excepts files to be in the same package

Reply:
Thinking whether it would have sense to copy the method (doesn't have to be exported) to the `upgrade` package.

Reply:
okay left as same as existing